### DISCUSSION TITLE
Can MBN information be extracted and parsed, AKA manually fixed?

### DISCUSSION THREAD

**Discussion by pipe2null** [iamromulan/cellular-modem-wiki#78] [Help - Quectel] [answered]
Problem with MBN from Probably-Last-Ever FW version RM521FGLEAR05A02M4G_01.200.01.200:

I am attempting to get both Data and SMS to work with a Verizon/Visible SIM.  At the end of the day, SMS is not really important to me, but if Verizon ever decides to force all companies who use their network to do SMS based 2FA, then I might be SOL.  The device I initially provisioned the SIM with was actually carrier locked (I didn't realize ahead of time), was able to complete the SIM setup but the phone is locked out from any SIM related upkeep.  My use of the SIM and modem should be just fine, but I don't want to deal with tech support should the need arise, and I'd much rather have a real solution in place than a hacky workaround.

The default MBN  "CDMAless-Verizon" has working SMS configuration with "ims" APN in slot#1, but I have not been able to get Data to work, even when nuking the "vzwadmin" APN, and manually activating the "vzwinternet" APN in slot iamromulan/cellular-modem-wiki#50.  Each activated and valid APN gets an IP, but the modem does not recognize it as the WAN connection, so no internet for the modem, no internet for downstream ethernet.  But SMS works.

If I switch to a different MBN "ROW_Commercial", I can manually created a "vzwinternet" APN in slot iamromulan/cellular-modem-wiki#51, and then internet works for all devices.  I can create "ims" in slot#2, and it will get an IP, but SMS does not work.

The various internet "gurus" I've found while searching conclude it impossible for both SMS and Data, or don't care beyond getting Data working and stop there.

I started a Quectel thread on the subject:
https://forums.quectel.com/t/verizon-sms-or-data-but-not-both/39756

Is  it possible to extract an MBN file and parse it to figure out the settings, and remap those settings to the standard "Data APN in slot#1" layout, and get SMS to work on slot#2 ?

**Comment by iamromulan** [comment, marked as answer]
Yes MBN can be messed with. Just found some open source tools but have not tried them. For Verizon though I did find a fix.
https://github.com/iamromulan/cellular-modem-wiki/blob/main/quectel/verizon_connection.md
